lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -420,6 +420,63 @@ void SPIRVRegularizeLLVMBase::cleanupConversionToNonStdIntegers(Module *M) {
   }
 }
 
+/// Move SPIRV builtin global variables to local memory
+/// Example: __spirv_BuiltInWorkgroupId is the builtin GV
+/// @__spirv_BuiltInWorkgroupId = external dso_local local_unnamed_addr
+///                               addrspace(1) constant <3 x i64>, align 32
+/// define weak_odr dso_local spir_kernel void @foo() {
+/// ...
+/// %0 = load i64, ptr addrspace(1) @__spirv_BuiltInWorkgroupId, align 32
+/// ret void
+/// }
+/// Transformed to --->
+/// @__spirv_BuiltInWorkgroupId = external dso_local local_unnamed_addr
+///                               addrspace(1) constant <3 x i64>, align 32
+/// define weak_odr dso_local spir_kernel void @foo() {
+/// ...
+/// %1 = load <3 x i64>, ptr addrspace(1) @__spirv_BuiltInWorkgroupId, align 32
+/// %__local_SPIRV_Builtin* = alloca <3 x i64> addrspace(0), align 32
```

Comment:
Also, while there were some recent improvements in instruction ordering in the generated module, still, can we ensure, that OpVariable is in the very begging of the basic block? Otherwise this validation rule is being violated:
`All OpVariable instructions in a function must be in the first block in the function.`

Reply:
I do not explicitly control where the OpVariable defined. in all my testing, I do see it occurring at the top.